### PR TITLE
1.2 versioning

### PR DESCRIPTION
**Pull request by ericstj**
This change increments the minor of all package versions and increments the build of any libraries that require it (to differentiate from 1.1 branch).  I've also set the package baseline to master.

In doing this I had to fix a few tests so that they were referencing the same live assembly build as the library they were testing.  I also cleaned up some pkgproj references that had unnecessary meta-data.

/cc @weshaggard @joperezr

### CODE REVIEW

File: src/System.Xml.ReaderWriter/dir.props [hunk 1]
```diff
@@ -1,7 +1,7 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
 <Project ToolsVersion="12.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <Import Project="..\dir.props" />
   <PropertyGroup>
-    <AssemblyVersion>4.1.0.0</AssemblyVersion>
+    <AssemblyVersion>4.2.0.0</AssemblyVersion>
```
Comment (bartonjs) [inline src/System.Xml.ReaderWriter/dir.props]: (Almost?) everywhere else you changed the build number, but here you changed the minor...

**Comment by joperezr** [comment]
> (to differentiate from 1.1 branch)

I get why we want to do the change, but just to be clear, prerelease tag should already be different between these two branches right?

File: pkg/Microsoft.NETCore.Platforms/Microsoft.NETCore.Platforms.pkgproj [hunk 2]
```diff
@@ -3,7 +3,7 @@
   <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.props))\dir.props" />
   
   <PropertyGroup>
-    <PackageVersion>1.0.2</PackageVersion>
+    <PackageVersion>1.2.0</PackageVersion>
```
Comment (weshaggard) [inline pkg/Microsoft.NETCore.Platforms/Microsoft.NETCore.Platforms.pkgproj]: Should we just use `$(PlatformPackageVersion)` here?

Comment (ericstj) [inline pkg/Microsoft.NETCore.Platforms/Microsoft.NETCore.Platforms.pkgproj]: I made a similar suggestion to @chcosta, he said he had trouble with it.  I can see if I can make that change.

**Comment by ericstj** [comment]
> I get why we want to do the change, but just to be clear, prerelease tag should already be different between these two branches right?

Yeah: that's the package.  We want the binaries to be different as well.  In particular we always bump the 3rd portion for bugfixes so that the assembly is unique and installs to a different location in the GAC.  This protects folks should someone decide to install it in the GAC and lets us uniquely target it for servicing fixes via publisher policy.

File: pkg/Microsoft.NETCore.Platforms/Microsoft.NETCore.Platforms.pkgproj (hunk 2 above)
Comment (ericstj) [inline pkg/Microsoft.NETCore.Platforms/Microsoft.NETCore.Platforms.pkgproj]: Done.

**Comment by ericstj** [comment]
Fixes https://github.com/dotnet/corefx/issues/12167

**Comment by ericstj** [comment]
@dotnet-bot test  Innerloop OSX Debug Build and Test

**Comment by ericstj** [comment]
@dotnet-bot test  Innerloop OSX Release Build and Test

**Comment by ericstj** [comment]
Fixed another conflict, hopefully will be able to merge 🙏

**Comment by ericstj** [comment]
@dotnet-bot test Innerloop OSX Debug Build and Test

**Pull request merged**